### PR TITLE
JAMES-2142 Add search in attachments in JMAP

### PR DESCRIPTION
**Pull request by aduprat**
Only JAMES-2142 commits: 5 last commits.

IT are not running on Cassandra, I don't have time to find why.
I commented all other test in GetMessageList integration test, only for fastest development.

### CODE REVIEW

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public MimePart build() {
             Optional<ParsedContent> parsedContent = parseContent(textExtractor);
             return new MimePart(
                 headerCollectionBuilder.build(),
-                parsedContent.map(ParsedContent::getTextualContent),
+                parsedContent.map(ParsedContent::getTextualContent).map(Optional::get),
```
Comment (chibenwa) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java]: IMO MimePart should accept an Optional

(Otherwize you throw everytime textual content is empty)

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java [hunk 2]
```diff
@@ -256,11 +256,14 @@ private boolean isInAttachments(String value, List<MessageAttachment> attachment
 
     private Stream<String> toAttachmentContent(Attachment attachment) {
         try {
-            return Stream.of(textExtractor
+            Stream.Builder<String> streamBuilder = Stream.builder();
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java]:
```
return OptionalUtils.toStream(
    textExtractor
                 .extractContent(
                     attachment.getStream(),
                     attachment.getType())
                .getTextualContent());
```

Seems to fit better team standards, and avoids side effects.

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java (hunk 1 above)
Comment (quynhn) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java]: parsedContent.map(ParsedContent::getTextualContent).orElse(Optional.empty())
Such as MimePart already accept bodyTextContent as Optional?

**Comment by aduprat** [comment]
#968

**Pull request closed without merging**